### PR TITLE
[Improvement](multi-catalog)Support refresh external catalog.

### PR DESCRIPTION
**Pull request by Jibing-Li**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Support manually refresh external catalog metadata.
1. refresh catalog external_catalog_name
2. refresh database catalog.db OR refresh database db (current catalog)
3. refresh table catalog.db.table OR refresh table db.table (current catalog) OR refresh table table_name (current db)

And the refresh operations above keep the database and table ids unchanged.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
5. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
6. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
7. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
8. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/datasource/HMSExternalCatalog.java [hunk 1]
```diff
@@ -84,10 +83,21 @@ private void init() {
             return;
         }
         for (String dbName : allDatabases) {
-            long dbId = Env.getCurrentEnv().getNextId();
-            dbNameToId.put(dbName, dbId);
-            idToDb.put(dbId, new HMSExternalDatabase(this, dbId, dbName));
+            long dbId;
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/HMSExternalCatalog.java:86]: Can you also modify `EsExternalCatalog` init logic?

Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/HMSExternalCatalog.java:86]: How about in next pr?

File: fe/fe-core/src/main/java/org/apache/doris/catalog/external/HMSExternalDatabase.java [hunk 2]
```diff
@@ -66,14 +66,32 @@ private synchronized void makeSureInitialized() {
...
         List<String> tableNames = extCatalog.listTableNames(null, name);
         if (tableNames != null) {
+            Map<String, Long> tmpTableNameToId = Maps.newConcurrentMap();
+            Map<Long, HMSExternalTable> tmpIdToTbl = Maps.newHashMap();
             for (String tableName : tableNames) {
-                long tblId = Env.getCurrentEnv().getNextId();
-                tableNameToId.put(tableName, tblId);
-                idToTbl.put(tblId, new HMSExternalTable(tblId, tableName, name, (HMSExternalCatalog) extCatalog));
+                long tblId;
+                if (tableNameToId != null && tableNameToId.containsKey(tableName)) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/external/HMSExternalDatabase.java:73]: `tableNameToId != null` is always true?

File: fe/fe-core/src/main/java/org/apache/doris/datasource/HMSExternalCatalog.java [hunk 3]
```diff
@@ -84,10 +83,21 @@ private void init() {
...
+            if (dbNameToId != null && dbNameToId.containsKey(dbName)) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/HMSExternalCatalog.java:87]: `dbNameToId != null` is always true?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/RefreshDbStmt.java [hunk 4]
```diff
@@ -34,16 +34,26 @@
 public class RefreshDbStmt extends DdlStmt {
     private static final Logger LOG = LogManager.getLogger(RefreshDbStmt.class);
 
+    private String catalogName;
```
[the remark this answers is not in the document]
Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/RefreshDbStmt.java:37]: Is it a problem? catalogName could be null, for example when use execute 
`refresh database db1`
I think the caller of getCatalogName should handle the null value.

File: fe/fe-core/src/main/java/org/apache/doris/datasource/HMSExternalCatalog.java (hunk 1 above)
Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/HMSExternalCatalog.java:86]: I will modify `EsExternalCatalog` in next commit.

File: fe/fe-core/src/main/java/org/apache/doris/analysis/RefreshDbStmt.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/RefreshDbStmt.java:37]: return null will be very error-prone. we can fill this field in analysis phase, to use default catalog saved in connection context.

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.55 seconds
 load time: 601 seconds
 storage size: 17107956263 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221018160341_clickbench_pr_30794.html

**Review by morningman — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
Please add regress test after #13143 is merged

**Pull request merged**